### PR TITLE
Parse extends and merge refactor

### PR DESCRIPTION
**Pull request by podviaznikov**
Now we have new function: `parseAndMergeMultiple`.

Which works exactly like parse, but takes multiple files and produces the result in the same format as `parse`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +32,10 @@ The response correspond to the following schema:
 			name: String, // Name specified for service in `docker-compose.yml`
 			isMain: Boolean, // Wether this should be the instance for which a repo instance should be created
 		},
+    extends: {
```
Comment (thejsj) [inline README.md:35]: Nit: this seems to not be aligned correctly

File: lib/index.js [hunk 2]
```diff
@@ -40,10 +40,12 @@ module.exports = class DockerComposeParser {
    *
    * @param {Object} opts
    * @param {String} opts.dockerComposeFileString - YML File string
+   * @param {String} opts.dockerComposeFilePath - Path to the Compose file
    * @param {String} opts.repositoryName - Name of repository in GH or other service
    * @param {String} opts.userContentDomain - User content domain for API environment. Used for creating hostnames.
    * @param {String} opts.ownerUsername - Organization name in Runnable. Used for creating hostnames.
    * @param {String} opts.scmDomain - Domain for used SCM. E.g. github.com
+   * @param {Boolean} opts.skipMissingMainCheck - skip check for miissing main
```
Comment (thejsj) [inline lib/index.js]: Nit: `miissing` > `missing`

File: lib/index.js [hunk 3]
```diff
@@ -71,6 +77,130 @@ module.exports = class DockerComposeParser {
...
+        dockerComposeFilePath: file.dockerComposeFilePath,
+        skipMissingMainCheck: true
+      })
+      return DockerComposeParser.parse(newOpts)
+    })
+    .then((parsedResults) => {
+      const allServices = parsedResults.reduce(
+        (servs, item) => {
+          servs = servs.concat(item.results)
+          return servs
```
Comment (thejsj) [inline lib/index.js]: Nit: `(servs, item) => servs.concat(item.results)`

**Review by thejsj — APPROVED** [review]

File: lib/index.js [hunk 4]
```diff
@@ -71,6 +77,130 @@ module.exports = class DockerComposeParser {
   }
 
   /**
+   * Parse a list if docker-compose.yml files into an array of services.
```
Comment (anandkumarpatel) [inline lib/index.js]: `s/if/of`

File: lib/index.js [hunk 5]
```diff
@@ -71,6 +77,130 @@ module.exports = class DockerComposeParser {
...
+   * @param {String} opts.userContentDomain - User content domain for API environment. Used for creating hostnames.
+   * @param {String} opts.ownerUsername - Organization name in Runnable. Used for creating hostnames.
+   * @param {String} opts.scmDomain - Domain for used SCM. E.g. github.com
+   * @resolves {Array<Object>}   result.results  - Array with objects for all services where `inheritance` was resolved by merging
+   * @resolves {Array<String>}   result.envFiles - Array with all required ENV files
+   */
+  static _mergeServices (allServices, opts) {
+    // services that have extend should be at the end
+    const sortedServices = allServices.sort((service) => {
+      if (!service.extends || !service.extends.service) {
```
Comment (anandkumarpatel) [inline lib/index.js]: nit: might be better as 
```
return (!service.extends || !service.extends.service) ? -1 : 1
```

File: lib/index.js [hunk 6]
```diff
@@ -71,6 +77,130 @@ module.exports = class DockerComposeParser {
...
+        return -1
+      }
+      return 1
+    })
+    // merge extended services with corresponding parents
+    const services = sortedServices.reduce(
+      (servs, service) => {
+        // if no `extends` just push it to the result and move on
+        if (!service.extends || !service.extends.service) {
```
Comment (anandkumarpatel) [inline lib/index.js:160]: no keypather?

**Review by anandkumarpatel — APPROVED** [review]

File: lib/index.js (hunk 6 above)
Comment (podviaznikov) [reply lib/index.js:160]: we don't use keypather in octobear, didn't want to add new dependency

**Comment by podviaznikov** [comment]
1__podviaznikov_anton-pro____runnable_octobear__zsh_

**Pull request merged**